bitrepository-core/src/main/java/org/bitrepository/common/utils/SettingsUtils.java

```diff
@@ -101,41 +101,51 @@ public static String getCollectionName(String collectionID) {
         return name;
     }
 
+    public static IntegrityServiceSettings getIntegrityServiceSettings() {
+        return settings.getReferenceSettings().getIntegrityServiceSettings();
+    }
+
+    /**
+     * Grab all PillarDetails from the ReferenceSettings if any are defined and return them as a list.
+     * Return an empty list if not defined.
+     * @return List of all PillarDetails defined in the ReferenceSettings
+     */
+    public static List<PillarIntegrityDetails.PillarDetails> getPillarIntegrityDetails() {
+        PillarIntegrityDetails pillarIntegrityDetails = getIntegrityServiceSettings().getPillarIntegrityDetails();
+        if (pillarIntegrityDetails != null) {
```

Comment:
Out of curiosity, this may be coded using `Optional`:
```
        return Optional.ofNullable(pillarIntegrityDetails)
                .map(PillarIntegrityDetails::getPillarDetails)
                .orElse(List.of());
```
or even
```
        return Optional.of(getIntegrityServiceSettings())
                .map(IntegrityServiceSettings::getPillarIntegrityDetails)
                .map(PillarIntegrityDetails::getPillarDetails)
                .orElse(List.of());
```
It certainly isn’t the primarily intended use of `Optional`, and some may consider it misuse, though. The code is fine as it stands.

Reply:
I'm not that big a fan of `Optional`s, but yeah one could do something like this. I think I prefer it as is.

Reply:
I agree. While a fan of `Optional` when used right, this would not have been the right use.